pronto/api/entry/signatures.py

```diff
@@ -385,3 +387,19 @@ def get_signatures_annotations(accession):
     cur.close()
     con.close()
     return jsonify(signatures)
+
+
+def check_annotation(acclist, cur):
+    search_acc = ['%' + acc + '%' for acc in acclist]
+    cur.execute(f"""SELECT A.ANN_ID, max(S.CT)
+    FROM INTERPRO.COMMON_ANNOTATION A
+    INNER JOIN INTERPRO.ENTRY2COMMON E ON A.ANN_ID = E.ANN_ID
+    LEFT OUTER JOIN (SELECT ANN_ID, COUNT(*) CT
+    FROM INTERPRO.ENTRY2COMMON GROUP BY ANN_ID) S ON A.ANN_ID = S.ANN_ID
+    WHERE {' OR '.join(["A.TEXT LIKE :1"] * len(acclist))}
+    GROUP BY A.ANN_ID""""", search_acc)
```

Comment:
What we want is the unique IDs for annotations where the query accessions are found. The avoid having duplicates to remove, we can avoid doing a `JOIN` between `COMMON_ANNOTATION` and `ENTRY2COMMON`, and instead ensuring that we only consider annotations found in `ENTRY2COMMON`:

```suggestion
    cur.execute(
        f"""
        SELECT ANN_ID
        FROM INTERPRO.COMMON_ANNOTATION
        WHERE ANN_ID IN (
            SELECT DISTINCT ANN_ID FROM INTERPRO.ENTRY2COMMON
        )
        AND ({' OR '.join(["TEXT LIKE :1"] * len(search_acc))})
        """,
        search_acc
    )
```